src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -849,86 +817,49 @@ private void createAndSetStatistics()
-        final String numberOfbakers = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.bakers", bakers);
-        final String numberOfcooks = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.cooks", cooks);
+                LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.totalCitizens", citizensSize, townHall.getColony().getMaxCitizens());
 
         final DecimalFormat df = new DecimalFormat("#.#");
         df.setRoundingMode(RoundingMode.CEILING);
         final String roundedHappiness = df.format(building.getColony().getOverallHappiness());
 
         findPaneOfTypeByID(HAPPINESS_LABEL, Label.class).setLabelText(roundedHappiness);
         findPaneOfTypeByID(TOTAL_CITIZENS_LABEL, Label.class).setLabelText(numberOfCitizens);
-        findPaneOfTypeByID(UNEMP_CITIZENS_LABEL, Label.class).setLabelText(numberOfUnemployed);
-        findPaneOfTypeByID(BUILDERS_LABEL, Label.class).setLabelText(numberOfBuilders);
-        findPaneOfTypeByID(DELIVERY_MAN_LABEL, Label.class).setLabelText(numberOfDeliverymen);
-        findPaneOfTypeByID(MINERS_LABEL, Label.class).setLabelText(numberOfMiners);
-        findPaneOfTypeByID(FISHERMEN_LABEL, Label.class).setLabelText(numberOfFishermen);
-        findPaneOfTypeByID(GUARDS_LABEL, Label.class).setLabelText(numberOfGuards);
-        findPaneOfTypeByID(LUMBERJACKS_LABEL, Label.class).setLabelText(numberOfLumberjacks);
-        findPaneOfTypeByID(FARMERS_LABEL, Label.class).setLabelText(numberOfFarmers);
-        findPaneOfTypeByID(BAKERS_LABEL, Label.class).setLabelText(numberOfbakers);
-        findPaneOfTypeByID(COOKS_LABEL, Label.class).setLabelText(numberOfcooks);
+
+        final Group group = findPaneOfTypeByID("citizen-stats", Group.class);
+        final Integer unemployed = jobCountMap.get("") == null ? 0 : jobCountMap.get("");
+        jobCountMap.remove("");
+
+        final Label unemployedLabel = new Label();
+        unemployedLabel.setSize(STATISTICS_LABEL_WIDTH, STATISTICS_LABEL_HEIGHT);
+        unemployedLabel.setTextAlignment(Alignment.MIDDLE_LEFT);
+        unemployedLabel.setColor(BLACK, BLACK);
+        final String numberOfUnemployed = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.unemployed", unemployed);
+        unemployedLabel.setLabelText(numberOfUnemployed);
+        group.addChild(unemployedLabel);
+
+        for (final Map.Entry<String, Integer> entry : jobCountMap.entrySet())
+        {
+            final Label workerLabel = new Label();
+            workerLabel.setSize(STATISTICS_LABEL_WIDTH, STATISTICS_LABEL_HEIGHT);
+            workerLabel.setTextAlignment(Alignment.MIDDLE_LEFT);
+            workerLabel.setColor(BLACK, BLACK);
+            final String job = entry.getKey();
+            final String labelJobKey = ( job.endsWith("man") ? job.replace("man", "men") : job + "s" );
```

Comment:
Ah nice, even with correct grammar =P

Reply:
Lol. I didn't want to change the i18n files. However, I think this will have a bug for one of the jobs because it had a capital letter in the properties file. I'll have to check when I return home.